src/Raven.Server/Web/System/OngoingTasksHandler.cs

```diff
@@ -123,14 +123,28 @@ private static IEnumerable<OngoingTask> CollectExternalReplicationTasks(List<Ext
                     };
                 }
 
+                string dstUrl = null;
+                if (tag == store.NodeTag)
+                {
+                    try
+                    {
+                        store.DatabasesLandlord.DatabasesCache.TryGetValue(name, out var task);
+                        dstUrl = task.Result.ReplicationLoader.OutgoingConnections.Single(o => o is ExternalReplication ex && ex.TaskId == watcher.TaskId).Url;
```

Comment:
Okay, you don't get to use `Single` any more.
There are multiple different failure conditions here that you are hiding with the `try`/ `catch`  which should be handled.

Reply:
yea, but it's only CollectExternalReplicationTasks method ... Should't I care here only about getting the url and ignoring rest of the issues?

Reply:
Bad error handling can cause a lot of hair pulling, let's avoid it.

Reply:
In particular, an error here in this manner may mean no replication for no observable reason